### ISSUE TITLE
Use "analytic=True" will lead to memory problem.

### ISSUE THREAD

**Issue by CoastEgo** [CoastEgo/microlux#1]
When using `optimistix` with `microlux` for event modeling. Setting `analytic=True` in `microlux.binary_mag` or `microlux.extended_light_curve` will lead to incredible memory usage.  

This is probably related to the current implementation of the analytical gradient and the design of the dynamic array (allocate too much memory).

We should add a warning on the docs and update the implementation.

**Comment by CoastEgo** [comment]
This is related to the reverse mode used in `optimistix` while the reverse mode in `microlux` is not preferred.
This can be fixed with the `custom_jvp` rule and use `jax.jacfwd` to force the forward mode differentiation.
